keras/utils/training_utils.py

```diff
@@ -155,13 +153,13 @@ def get_slice(data, i, parts):
                     outputs = [outputs]
 
                 # Save the outputs for merging back together later.
-                for o in range(len(outputs)):
-                    all_outputs[o].append(outputs[o])
+                for name, o in zip(model.output_names, outputs):
+                    all_outputs.setdefault(name, []).append(o)
 
     # Merge outputs on CPU.
     with tf.device('/cpu:0'):
         merged = []
-        for outputs in all_outputs:
+        for name, outputs in all_outputs.items():
```

Comment:
This is no longer ordered, which I think is not intuitive.

Simpler: don't change anything until this line (i.e. use a list of lists), and at this line, use `zip(model.output_names, all_outputs)`.

Reply:
Good point.